src/context/internal.rs

```diff
@@ -126,22 +126,24 @@ pub trait ContextInternal<'a>: Sized {
     fn deactivate(&self) { self.scope_metadata().active.set(false); }
 
     #[cfg(feature = "legacy-runtime")]
-    fn try_catch_internal<'b: 'a, T, F>(&mut self, f: F) -> Result<Handle<'a, T>, Handle<'a, JsValue>>
-        where T: Value,
-              F: UnwindSafe + FnOnce(&mut Self) -> JsResult<'b, T>
+    fn try_catch_internal<'b: 'a, T, F>(&mut self, f: F) -> Result<T, Handle<'a, JsValue>>
+        where T: Sized,
+              F: UnwindSafe + FnOnce(&mut Self) -> NeonResult<T>,
     {
         // A closure does not have a guaranteed layout, so we need to box it in order to pass
         // a pointer to it across the boundary into C++.
         let rust_thunk = Box::into_raw(Box::new(f));
 
-        let mut local: MaybeUninit<raw::Local> = MaybeUninit::zeroed();
+        let mut ok: MaybeUninit<T> = MaybeUninit::zeroed();
+        let mut err: MaybeUninit<raw::Local> = MaybeUninit::zeroed();
         let mut unwind_value: MaybeUninit<*mut c_void> = MaybeUninit::zeroed();
 
         let ctrl = unsafe {
             neon_runtime::try_catch::with(try_catch_glue::<Self, T, F>,
                                           rust_thunk as *mut c_void,
                                           (self as *mut Self) as *mut c_void,
-                                          local.as_mut_ptr(),
+                                          ok.as_mut_ptr() as *mut c_void,
```

Comment:
I think if there were problems around Sized, this cast would fail, since Rust `as` casts only allow casting between pointers of the same size.

Reply:
Okay, I found the citation for you: https://doc.rust-lang.org/reference/type-layout.html#pointers-and-references-layout

> Pointers to sized types have the same size and alignment as `usize`.

Reply:
Awesome, thanks!